stdlib/public/Concurrency/Task.cpp

```diff
@@ -332,11 +362,16 @@ static void task_wait_throwing_resume_adapter(SWIFT_ASYNC_CONTEXT AsyncContext *
 }
 
 /// All `swift_task_create*` variants funnel into this common implementation.
+///
+/// If \p isAsyncLetTask is true, the \p closureContext is not heap allocated,
+/// but stack-allocated (and must not be ref-counted).
+/// Also, async-let tasks are not heap allcoated, but allcoated with the parent
+/// task's stack allocator.
 static AsyncTaskAndContext swift_task_create_group_future_commonImpl(
     JobFlags flags, TaskGroup *group,
     const Metadata *futureResultType,
     FutureAsyncSignature::FunctionType *function,
-    void *closureContext, bool owningClosureContext,
+    void *closureContext, bool isAsyncLetTask,
     size_t initialContextSize) {
```

Comment:
ABI wise my impression was we wanted something like `[TaskCreateRecords]` to be passed in and e.g. check their `.kind == .isAsyncLet` or similar to be future proof about adding more smarts to creating tasks.

But maybe we want to do that in a separate PR? (totally find by me to do later on!)

Reply:
yeah, let's do that in a separate PR

Reply:
okey 👍